src/form.js

```diff
@@ -308,6 +309,11 @@ define([
                         meta = null;
                     }
                     this.knownInstances[attrs.id] = attrs.src;
+                } else if (meta.attributes.id !== attrs.id) {
+                    // support renaming when there is no reference
```

Comment:
Not sure if this is referring to instances added with "Instance ID" and "Instance URI" fields in model iteration? FWIW, those fields should be removed since we now get a list of known data sources from HQ and no longer need to have the user enter those details manually. Any sufficiently advanced user who wants to add an instance that is not available in the list of data sources provided by HQ can manually edit XML to do so.